src/main/java/com/ezasm/simulation/Simulator.java

```diff
@@ -113,6 +115,8 @@ public void addLines(Collection<Line> content) {
     public void runLine(Line line) throws SimulationException {
         if (line != null && !line.isLabel()) {
             instructionDispatcher.execute(line);
+            // Highlight changed Value
+            Window.highlightValue(line.getRegisterNum());
```

Comment:
This should not be in the Simulator but rather the SimulationGUI class which handles GUI simulations since this is solely a GUI feature.